File: ios/Cohort/Services/VerificationService.swift
//
//  VerificationService.swift
//  Cohort
//

import Foundation

enum VerificationOutcome: String, Codable {
    case success
    case failure
    case manualReview = "manual-review"
}

struct VerificationResponse {
    let status: VerificationOutcome
    let message: String?
}

actor VerificationService {
    func submitVerificationPhoto(photoData: Data) async throws -> VerificationResponse {
        try await Task.sleep(nanoseconds: 500_000_000)
        return VerificationResponse(status: .success, message: nil)
    }
}

File: src/app/services/verificationService.ts
export type VerificationOutcome = 'success' | 'failure' | 'manual-review';

export interface VerificationResponse {
  status: VerificationOutcome;
  message?: string;
}

export async function submitVerificationPhoto(photoData: string): Promise<VerificationResponse> {
  if (!photoData) {
    throw new Error('We could not read your photo. Please try again.');
  }

  await Promise.resolve();

  return {
    status: 'success',
  };
}
